docs/testplan/pnac-test-plan.md

```diff
@@ -0,0 +1,455 @@
+
+"6"\*            AKA'
+
+"7"\*            AKA'
+
+"8"\*            AKA'
+
+\# Wildcard for all other identities
+
+\*               PEAP,TTLS,TLS,SIM,AKA
+
+\# Phase 2 (tunnelled within EAP-PEAP or EAP-TTLS) users
+
+"t-md5"         MD5     "password"      [2]
+
+"DOMAIN\t-mschapv2"     MSCHAPV2        "password"      [2]
+
+"t-gtc"         GTC     "password"      [2]
+
+"not anonymous" MSCHAPV2        "password"      [2]
+
+"user"          MD5,GTC,MSCHAPV2        "password"      [2]
+
+"test user"     MSCHAPV2        hash:000102030405060708090a0b0c0d0e0f   [2]
+
+"ttls-user"     TTLS-PAP,TTLS-CHAP,TTLS-MSCHAP,TTLS-MSCHAPV2    "password"
+
+[2]
+
+\# Default to EAP-SIM and EAP-AKA based on fixed identity prefixes in phase 2
+
+"0"\*            AKA     [2]
+
+"1"\*            SIM     [2]
+
+"2"\*            AKA     [2]
+
+"3"\*            SIM     [2]
+
+"4"\*            AKA     [2]
+
+"5"\*            SIM     [2]
+
+"6"\*            AKA'    [2]
+
+"7"\*            AKA'    [2]
+
+"8"\*            AKA'    [2]
+
+**Test Case (2.1): Port based NAC (local authentication) on an interface**
+
+**Test Objective:** To verify the Port based NAC (local authentication) on an interface using 
+
+`                              `a valid password. 
+
+**Test Steps:** 1. Log in to the supplicant machine and enter the supplicant file path.
+
+`                       `2. Verify the credential details mentioned in the “wpa\_supplicant.conf” file.
+
+`                       `3. Log in to the switch with preloaded PNAC supported SONiC image.
+
+`                       `4. Verify if the NAC status is UP in global mode and on the specific NAC 
+
+`                            `interface.
+
+`                       `5. Start the supplicant and verify that the authentication is successful.
+
+`           `6. Verify whether the NAC enabled interface is authorized.
+
+`                       `7. Send traffic from supplicant and verify that the traffic is received on the 
+
+`                           `network PC.
+
+**Test Case (2.2): Port based NAC (local authentication) on an interface**
+
+**Test Objective:** To verify the Port based NAC (local authentication) on an interface using 
+
+`                              `an invalid password.
+
+**Test Steps:** 1. Log in to the supplicant machine and enter the supplicant file path.
+
+`                       `2. Verify the credential details mentioned in the “wpa\_supplicant.conf” file.
+
+`                       `3. Log in to the switch with preloaded PNAC supported SONiC image.
+
+`                       `4. Verify if the NAC status is UP in global mode and on the specific NAC
+
+`                           `interface.
+
+`                       `5. Start the supplicant and ensure that the authentication fails.
+
+`           `6. Verify if the NAC enabled interface is unauthorized.
+
+`                       `7. Verify that the network PC does not receive any traffic from the supplicant
```

Comment:
Please give some details about what application is being used on the PC for 802.1x Supplicant and how.

Reply:
Used (https://www.linuxfromscratch.org/blfs/view/10.1-systemd/basicnet/wpa_supplicant.html). 
  (HostApd Version - (Version used – 2.9) – Latest Version is 2.10)